docs/using-the-nodejs-wrapper/using-plugins/UsingTheFailover2Plugin.md

```diff
@@ -0,0 +1,80 @@
+
+## Differences between the Failover Plugin and the Failover Plugin v2
+
+The Failover Plugin performs a failover process for each DB connection. Each failover process is triggered independently and is unrelated to failover processes in other connections. While such independence between failover processes has some benefits, it also leads to additional resources like extra asynchronous tasks. If dozens of DB connections are failing over at the same time, it may cause significant load on a client environment.
+
+<div style="text-align:center"><img src="../../images/failover1.png"  alt="failover diagram"/></div>
+<i>Picture 1. Each connection triggers its own failover process to detect a new writer.</i>
+<br><br>
+
+If a connection needs to get the latest topology, it calls `RdsHostListProvider`. As shown in _Picture 1_ above, different connections start and end their failover processes independently.
+
+The Failover Plugin v2 uses an optimized approach where the process of detecting and confirming a cluster topology is delegated to a central topology monitoring component that runs a separate asynchronous monitoring task. When the topology is confirmed and a new writer is detected, each waiting connection can resume and reconnect to a required host. This design helps minimize resources required for failover processing and scales better compared to the Failover Plugin.
+
+<div style="text-align:center"><img src="../../images/failover2.png"  alt="failover2 diagram"/></div>
+<i>Picture 2. Connections call MonitoringRdsHostListProvider, which is responsible for detecting the new writer.</i>
+<br><br>
+
+If two connections encounter communication issues with their internal (physical) DB connections, each connection may send a request to the topology monitoring component (`MonitoringRdsHostListProvider` in _Picture 2_) for updated topology information reflecting the new writer. Both connections are notified as soon as the latest topology is available. Connection tasks can resume, continue with their suspended workflows, and reconnect to a reader or a writer host as needed.
+
+The topology monitoring component mentioned above (`MonitoringRdsHostListProvider`) fetches and confirms topology periodically. Outside of failover it performs "slow rate monitoring" at intervals set by `clusterTopologyRefreshRateMs`. This ensures use of the most up to date topology. When failover is triggered, it enters "panic mode" and works to detect a writer. After a new writer has been detected, the `MonitoringRdsHostListProvider` enters "high rate monitoring", with intervals set by `clusterTopologyHighRefreshRateMs`. This allows time for all readers to appear in the topology. The default of 30 seconds is usually enough time for cluster failover to complete and cluster topology to stabilize.
```

Comment:
check that `clusterTopologyHighRefreshRateMs` is the name in the wrapper properties